### DISCUSSION TITLE
Number of threads

### DISCUSSION THREAD

**Discussion by betsch85** [uNetworking/uWebSockets#1762] [Q&A] [answered]
Hi,

in `READMORE.MD` you write: "The library is single threaded."

yet, as soon as the App starts listening, it spawns ~20+ threads. When I pause a thread, the debugger jumps to 
```
void run() {
    us_loop_run((us_loop_t *) this);
}
```

in `Loop.h`

Is there a way to reduce the amount of threads being created? I have to deal with one or two connections tops, so 20+ threads is kind of overkill

Thanks

**Comment by uNetworkingAB** [comment, marked as answer]
What are you even talking about? You have it 100% wrong.

**Reply by betsch85** [reply]
yeah, you're right. Got tricked by VSCode C++ and GDB

No matter which thread I paused, VSCode jumped to the current line that was executed in the main thread, visually it seemed like it jumped to the current line in the actual thread I've paused
